src/lib/services/nft-collection/index.ts

```diff
@@ -40,23 +36,19 @@ export const useCollections = (
   limit: number,
   offset: number,
   search?: string,
-  options?: Pick<UseQueryOptions<CollectionsResponse>, "onSuccess">
+  options?: Pick<
+    UseQueryOptions<NftCollectionsResponse>,
+    "onSuccess" | "enabled"
+  >
 ) => {
-  const { chainConfig } = useCelatoneApp();
-  return useQuery<CollectionsResponse>(
-    [
-      CELATONE_QUERY_KEYS.NFT_COLLECTIONS,
-      chainConfig.graphql,
-      limit,
-      offset,
-      search,
-    ],
-    async () =>
-      getCollections(chainConfig.graphql ?? "", limit, offset, search),
+  const apiEndpoint = useBaseApiRoute("nft_collections");
+
+  return useQuery(
+    [CELATONE_QUERY_KEYS.NFT_COLLECTIONS, limit, offset, search, apiEndpoint],
```

Comment:
reorder to `[CELATONE_QUERY_KEYS.NFT_COLLECTIONS, apiEndpoint, limit, offset, search],`